apps/fortuna/src/command/run.rs

```diff
@@ -121,38 +122,69 @@ pub async fn run_keeper(
 
 pub async fn run(opts: &RunOptions) -> Result<()> {
     let config = Config::load(&opts.config.config)?;
+    let provider_config = ProviderConfig::load(&opts.provider_config.provider_config)?;
     let private_key = opts.load_private_key()?;
     let secret = opts.randomness.load_secret()?;
     let (tx_exit, rx_exit) = watch::channel(false);
 
     let mut chains: HashMap<ChainId, BlockchainState> = HashMap::new();
     for (chain_id, chain_config) in &config.chains {
         let contract = Arc::new(PythContract::from_config(&chain_config)?);
+        let provider_config = provider_config.get_chain_config(chain_id)?;
```

Comment:
this is confusing naming. We have two `provider_config`s. Maybe the other one should be plural?

Reply:
yeah, right